service/src/main/java/bio/terra/externalcreds/dataAccess/SshKeyPairDAO.java

```diff
@@ -0,0 +1,81 @@
+package bio.terra.externalcreds.dataAccess;
+
+import bio.terra.externalcreds.generated.model.SshKeyPairType;
+import bio.terra.externalcreds.models.SshKeyPair;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Optional;
+import org.springframework.dao.support.DataAccessUtils;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class SshKeyPairDAO {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  public SshKeyPairDAO(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public Optional<SshKeyPair> getSshKeyPair(String userId, SshKeyPairType type) {
+    var namedParameters =
+        new MapSqlParameterSource().addValue("userId", userId).addValue("type", type.name());
+    var query = "SELECT * FROM ssh_key_pair WHERE user_id = :userId AND type = :type";
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(
+            jdbcTemplate.query(query, namedParameters, new SshKeyPairRowMapper())));
```

Comment:
You can make a static instance of the row mapper rather than `new`-ing it each time.
See [WSM ResourceDao](https://github.com/DataBiosphere/terra-workspace-manager/blob/54ff3d68d6407069054b4756ba38dede059da3a4/service/src/main/java/bio/terra/workspace/db/ResourceDao.java#L75) for an example.

Reply:
good point